react/src/lib/components/DeckGLMap/layers/LassoLayer/lassoLayer.tsx

```diff
@@ -0,0 +1,115 @@
+import { CompositeLayer, PickingInfo, LayersList } from "@deck.gl/core/typed";
+import { FeatureCollection } from "@nebula.gl/edit-modes";
+import { layersDefaultProps } from "../layersDefaultProps";
+import { SelectionLayer } from "@nebula.gl/layers";
+import UnfoldedGeoJsonLayer from "../intersection/unfoldedGeoJsonLayer";
+import { ExtendedLayerProps } from "../utils/layerTools";
+import { getSize } from "../wells/wellsLayer";
+import { Color } from "@deck.gl/core/typed";
+import { Feature } from "geojson";
+export interface LassoLayerProps<D> extends ExtendedLayerProps<D> {
+    mode: string; // One of modes in MODE_MAP
+    selectedFeatureIndexes: number[];
+    pickingInfos: PickingInfo[];
+    refine: boolean;
+    pointRadiusScale: number;
+    lineWidthScale: number;
+    lineStyle: LineStyleAccessor;
+    wellHeadStyle: WellHeadStyleAccessor;
+}
+
+type StyleAccessorFunction = (
+    object: Feature,
+    objectInfo?: Record<string, unknown>
+) => StyleData;
+
+type NumberPair = [number, number];
+type DashAccessor = boolean | NumberPair | StyleAccessorFunction | undefined;
+type ColorAccessor = Color | StyleAccessorFunction | undefined;
+type SizeAccessor = number | StyleAccessorFunction | undefined;
+type StyleData = NumberPair | Color | number;
+
+type LineStyleAccessor = {
+    color?: ColorAccessor;
+    dash?: DashAccessor;
+    width?: SizeAccessor;
+};
+
+type WellHeadStyleAccessor = {
+    color?: ColorAccessor;
+    size?: SizeAccessor;
+};
+
+// Composite layer that contains an Selection Lyaer from nebula.gl
+// See https://nebula.gl/docs/api-reference/layers/selection-layer
+export default class LassoLayer extends CompositeLayer<
+    LassoLayerProps<FeatureCollection>
+> {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    setMultiSelection(pickingInfos: any[]): void {
+        if (this.internalState) {
+            const data = pickingInfos
+                .map((item) => item.object)
+                .filter((item) => item.type === "Feature");
+            this.setState({
+                pickingInfos: pickingInfos,
+                data: data,
+            });
+        }
+    }
+
+    renderLayers(): LayersList {
+        if (this.props.visible == false) {
+            return [];
+        }
+
+        const LINE = "line";
+        const POINT = "point";
+        const isOrthographic =
+            this.context.viewport.constructor.name === "OrthographicViewport";
+        const positionFormat = isOrthographic ? "XY" : "XYZ";
+
+        const geoJsonLayer = new UnfoldedGeoJsonLayer({
```

Comment:
The `UnfoldedGeoJsonLayer` is for "unfolded" or "section" views as far as I remember. I think you can use the standard `GeoJsonLayer`.